facilitator/src/idl.rs

```diff
@@ -58,6 +58,8 @@ lazy_static! {
 }
 
 pub trait Header: Sized {
+    /// Sets the SHA256 digest of the packet file this header describes.
+    fn set_packet_file_digest(&mut self, digest: impl Into<Vec<u8>>);
```

Comment:
This is slick, and I like that you're not forcing the caller to heap-allocate a `Vec` here. As elsewhere, I think making this `fn set_packet_file_digest<D: Into<Vec<u8>>(&mut self, digest: D)` is slightly better.